test/test.py

```diff
@@ -0,0 +1,70 @@
+"""Unit tests.
+
+TODO's
+ 1. CLI tests
+ 2. Python API tests
+ 3. CompLoincTest: Anything redundant to remove?
+ 4. later: Need to run sequentially? Prob not, but have SequentialTests from pma-api I could re-use
+ 5. later: Run from __main__? if so, pma-api has a python api to run tests
+ 6. How to test outputs? file size? existence? arbitrary content? md5 match?
+"""
+import os
+import subprocess
+import unittest
+
+
+try:
+    from test.config import PROJECT_DIR, TEST_STATIC_DIR
+except ModuleNotFoundError:
+    from config import PROJECT_DIR, TEST_STATIC_DIR
+
+
+class StaticFileTests(unittest.TestCase):
+    """super class for common test functions"""
+
+    # todo: May not need to use, since using static files at project root.
+    # @staticmethod
+    # def get_input_name_path_map(test_method_dir: str) -> Dict[str, Path]:
+    #     """Get list of files in a directory."""
+    #     dir_glob_path = os.path.join(TEST_STATIC_DIR, test_method_dir, 'input', '*')
+    #     paths: List[str] = glob(dir_glob_path)
+    #     return {
+    #         os.path.basename(path): path
+    #         for path in paths if not os.path.isdir(path)
+    #     }
+
+    @staticmethod
+    def outdir(test_method_dir: str) -> str:
+        """Get list of files in a directory."""
+        return os.path.join(TEST_STATIC_DIR, test_method_dir, 'output')
+
+
+class CompLoincTests(StaticFileTests):
+    """CompLOINC tests"""
+
+    cli_command_base = 'python comp_loinc/build.py'
+
+    def all(self):
+        """Wrapper for running directly in the debugger."""
+        self.test_cli_parts()
+
+    # TODO: codes, composed, merge, reason; all
+    def test_cli_parts(self):
+        """Test CLI: parts"""
+        test_name = 'test_cli_parts'
+        outpath = os.path.join(self.outdir(test_name), 'part_ontology.owl')
+        command_str = \
+            f'{self.cli_command_base} parts ' \
+            f'--schema-file ./model/schema/part_schema.yaml ' \
+            f'--part-directory ./data/part_files ' \
+            f'--output {outpath}'
+        command_list = command_str.split(' ')
+        os.remove(outpath)
+        subprocess.run(command_list, cwd=PROJECT_DIR)
+        size_kb = os.path.getsize(outpath) / 1000
+
+        self.assertGreaterEqual(size_kb, 500)  # arbitrary size for now
```

Comment:
@timbrisc Here are the options I thought of to test outputs:
a. **File size**: (arbitrarily checks if over a certain size; I'm looking at each file and lowering the required dize by a bit). This is easy and at least verifies output.
b. **md5**: exact match to known output: This would be great, assuming files didn't change, but they're going to be changed often.
c. **contents**: Could use oak to check some things, like maybe presence of some classes that we expect to never change.

I've chosen **(a)**.  **(c)** is probably the best general option, but takes more time to set up. With my very low effort, my recommendation is to go with (a) for now and move to (c) at some point in the future.